src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -64,6 +64,18 @@ public SyntaxToken ParseToken(string text)
         public SyntaxTriviaList ParseLeadingTrivia(string text)
             => SyntaxFactory.ParseLeadingTrivia(text);
 
+        public string EscapeIdentifier(string identifier)
+        {
+            var nullIndex = identifier.IndexOf('\0');
```

Comment:
when does thsi happen?? :(

Reply:
I'm not sure tbh. Came from http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/StringExtensions.cs,13 

I'm willing to remove in favor of simplicity, and if a bug happens then fix and document what's happening...